wasp-ai/src/client/pages/ResultPage.jsx

```diff
@@ -406,59 +413,71 @@ function getCardinalNumber(number) {
   }
 }
 
-export function StarOurRepoModal({ isStarRepoOpen, setIsStarRepoOpen, appGenerationResult }) {
-  const [tokens, setTokens] = useState(0)
-  const { data: numTotalProjects } = useQuery(getNumProjects, {}, { enabled: isStarRepoOpen })
-
-  const tokenNumberStr = appGenerationResult?.project?.logs?.filter(log => log.content.toLowerCase().includes("tokens usage") === true)[0]?.content.split(':')[1]?.trim()
+export function OnSuccessModal({ isOpen, setIsOpen, appGenerationResult }) {
+  const [numTokensSpent, setNumTokensSpent] = useState(0)
+  const { data: numTotalProjects } = useQuery(getNumProjects, {}, { enabled: isOpen })
 
   useEffect(() => {
-    if (!!tokenNumberStr) {
-      const num = tokenNumberStr.slice(1, -1) * 1000
-      setTokens(num)
+    const logText = appGenerationResult?.project?.logs?.find((log) => /tokens usage/i.test(log.content))?.content;
+    const tokenNumberStr = logText ? logText.split(":")[1]?.trim() : null;
```

Comment:
Can't you combine these two into one regex?
If I am correct, you are looking for something like `tokens usage: 25k`. You can go after that with one regex: `tokens\s+usage\s*:\s*([0-9]+)\s*k\b`. This should allow for more freedom with whitespace, so we are more robust, and it also extracts the number right away (puts it into the match group), I am not sure how exactly it works in JS but should be a relatively straightforward way to extract that match group, getting exactly the number out of it.

Btw I often use this page to test out regexes: https://regex101.com/ .

Reply:
Here's a piece of code as well

```js
const string = "📊 Total tokens usage: ~52.4k ";
const regex = /Total tokens usage: ~(\d+\.\d+?)k/;
const match = string.match(regex);

if (match) {
    const num = parseFloat(match[1]);
    console.log(num);
} else {
    console.log("No match found");
}
```

Reply:
Nice! This won't work for numbers that don't have `.` in them, like `52k` though.

Here is another iteration on the regex:
```
Total\s+tokens\s+usage\s*:\s*~\s*(\d+(?:\.\d+){0,1})\s*k\b
```
1. I added `\s+` or `\s*` instead of actual space, to make it more robust to changes in whitespace.
2. I enclosed `\.\d+` in `(?: ){0,1}` which says this sequence can occur zero or one time.
3. I added `\b` at the very end, so `k` won't match the case where you have `~52.4kilos`, instead `k` has to be end of the word, no other letters or digits or something after it -> but spaces, newlines, punctuation and whatever are ok. `\b` stands for boundary.